sqlmesh/core/snapshot/definition.py

```diff
@@ -1207,6 +1207,14 @@ def _ensure_categorized(self) -> None:
         if not self.version:
             raise SQLMeshError(f"Snapshot {self.snapshot_id} has not been versioned yet.")
 
+    def __getstate__(self) -> t.Dict[t.Any, t.Any]:
+        state = super().__getstate__()
+        state["__dict__"] = state["__dict__"].copy()
+        # Don't store intervals.
+        state["__dict__"]["intervals"] = []
```

Comment:
should we do this for copy too or no? we replicate these in models

Reply:
I've confirmed that we rely on the fact that intervals are preserved after copying. This is only relevant when pickling.